### PR TITLE
Less translation in parts

### PR DESCRIPTION
**Pull request by pafcu**
Try to not build up translation from partially translated sentences and also use pluralization provided by the i18n library.

### CODE REVIEW

File: src/components/views/login/InteractiveAuthEntryComponents.js [hunk 1]
```diff
@@ -256,7 +256,7 @@ export const EmailIdentityAuthEntry = React.createClass({
         } else {
             return (
                 <div>
-                    <p>{ _t("An email has been sent to") } <i>{ this.props.inputs.emailAddress }</i></p>
+                    <p>{ _t("An email has been sent to %(emailAddress)s", { emailAddress: '<i>' + this.props.inputs.emailAddress + '</i>' }) }</p>
```
Comment (t3chguy) [inline src/components/views/login/InteractiveAuthEntryComponents.js]: Won't this write the string `<i>`?

File: src/components/views/login/InteractiveAuthEntryComponents.js [hunk 2]
```diff
@@ -370,7 +370,7 @@ export const MsisdnAuthEntry = React.createClass({
             });
             return (
                 <div>
-                    <p>{ _t("A text message has been sent to") } +<i>{ this._msisdn }</i></p>
+                    <p>{ _t("A text message has been sent to %(msisdn)s", { msisdn: '<i>' + this._msisdn + '</i>' }) }</p>
```
Comment (t3chguy) [inline src/components/views/login/InteractiveAuthEntryComponents.js]: same as above, it looks like this'll need _tJsx

File: src/components/views/rooms/RoomSettings.js [hunk 3]
```diff
@@ -867,21 +868,21 @@ module.exports = React.createClass({
                                     disabled={!roomState.mayClientSendStateEvent("m.room.history_visibility", cli)}
                                     checked={historyVisibility === "shared"}
                                     onChange={this._onHistoryRadioToggle} />
-                            { _t('Members only') } ({ _t('since the point in time of selecting this option') })
+                            { _t('Members only since the point in time of selecting this option') })
```
Comment (t3chguy) [inline src/components/views/rooms/RoomSettings.js]: this is removing the brackets that were there previously, any reason?

File: src/components/views/rooms/RoomSettings.js [hunk 4]
```diff
@@ -867,21 +868,21 @@ module.exports = React.createClass({
...
                         </label>
                         <label>
                             <input type="radio" name="historyVis" value="invited"
                                     disabled={!roomState.mayClientSendStateEvent("m.room.history_visibility", cli)}
                                     checked={historyVisibility === "invited"}
                                     onChange={this._onHistoryRadioToggle} />
-                            { _t('Members only') } ({ _t('since they were invited') })
+                            { _t('Members only since they were invited') })
```
Comment (t3chguy) [inline src/components/views/rooms/RoomSettings.js]: this is removing the brackets that were there previously, any reason?

File: src/components/views/rooms/RoomSettings.js [hunk 5]
```diff
@@ -867,21 +868,21 @@ module.exports = React.createClass({
...
                         </label>
                         <label >
                             <input type="radio" name="historyVis" value="joined"
                                     disabled={!roomState.mayClientSendStateEvent("m.room.history_visibility", cli)}
                                     checked={historyVisibility === "joined"}
                                     onChange={this._onHistoryRadioToggle} />
-                            { _t('Members only') } ({ _t('since they joined') })
+                            { _t('Members only since they joined') })
```
Comment (t3chguy) [inline src/components/views/rooms/RoomSettings.js]: this is removing the brackets that were there previously, any reason?

File: src/components/views/login/InteractiveAuthEntryComponents.js (hunk 1 above)
Comment (MTRNord) [reply src/components/views/login/InteractiveAuthEntryComponents.js]: I think too there needs to be _tJsx used

**Comment by pafcu** [comment]
Problems found by @t3chguy should be fixed not. Thanks!

**Comment by lukebarnard1** [comment]
@pafcu I'm surprised there aren't many conflicts with this. Would you mind resolving them so that I can take a look?

**Comment by pafcu** [comment]
Updated

**Comment by lukebarnard1** [comment]
@pafcu looks like you'll need to run the translation generation script.

**Comment by pafcu** [comment]
If you mean the missing translations, it's because of a bug in the test scripts. See https://github.com/vector-im/riot-web/issues/5426

**Comment by pafcu** [comment]
Is this blocked on getting tests fixed?

**Comment by lukebarnard1** [comment]
@pafcu yep

**Comment by pafcu** [comment]
I would perhaps rather disable the test in this instance, in order to minimize wasted work by translators. Unless some fix is already being worked on.

**Comment by lukebarnard1** [comment]
When the tests are fixed, we can check that this PR really passes the tests and then merge, no work being wasted to my knowledge.

**Comment by pafcu** [comment]
This commit changes several strings which will potentially need to be retranslated once merged. Translating something twice is more work than translating it once. Also, a lot of extra mental work trying to figure out what the current strings mean, and how to translate them correctly.

**Comment by lukebarnard1** [comment]
So. Having fixed the tests (not PR'd yet, but will be) I've noticed that the tests would still fail legitimately because "one other" has become "1 others" and likewise "joined" has become "joined 1 times" , which is not what we want e.g.:

```
Error: Expected 'user_1, user_2 and 1 others joined 1 times' to be
                'user_1, user_2 and one other joined'
```

**Comment by lukebarnard1** [comment]
And now that https://github.com/vector-im/riot-web/issues/5426 has landed, the tests should pass.

**Pull request merged**